src/smb_file.c

```diff
@@ -222,7 +229,21 @@ ssize_t   smb_fread(smb_session *s, smb_fd fd, void *buf, size_t buf_size)
     if (!smb_session_check_nt_status(s, &resp_msg))
         return -1;
 
+    if (resp_msg.payload_size < sizeof(smb_read_resp))
+    {
+        BDSM_dbg("[smb_fread]Malformed message.\n");
+        return DSM_ERROR_NETWORK;
+    }
+
     resp = (smb_read_resp *)resp_msg.packet->payload;
+
+    if (resp_msg.packet->payload + resp_msg.payload_size <
+        (uint8_t *)resp_msg.packet + resp->data_offset + resp->data_len)
```

Comment:
Wouldn't it be easier to use sizeof rather than pointer comparisons ?

Reply:
I am afraid not.. Because `resp->data_offset + resp->data_len` may be larger than the result of sizeof(smb_read_resp), the real data will be lost.
The structure smb_read_resp does not contain the actual data of the file, so the result of sizeof operator does not contain the file data either.